app/assets/javascripts/discourse/helpers/category-link.js.es6

```diff
@@ -29,6 +29,10 @@ function categoryStripe(color, classes) {
 export function categoryBadgeHTML(category, opts) {
   opts = opts || {};
 
+  if (typeof category === "number") {
```

Comment:
I find this kind of API "surprising" and I think in the long term it's much easier to pick one rather than the other. I suggest only accepting the category object, and having the route wire it up for you. I should note if you use the Store for doing this query, category_id should automatically be converted to objects for you.

Reply:
Ok done, does it look good now?